packages/dev/gui/src/2D/controls/gradient/BaseGradient.ts

```diff
@@ -0,0 +1,95 @@
+/*

```

Comment:
Wondering out loud - 

it looks like all of the gradient classes are not more than a JSON disguised as a class. What I mean is - they mostly hold state data, but have little to no functionality publicly exposed. The one function that might be needed here is the `addColorStop` (and `remove...` of course), but those can be external functions. Look at the serialize and parse functions - they do nothing more than assigning the values to the right variables.

Note - this can be answered with "this architecture makes a lot more sense to me, let's keep it this way", and I'll totally accept that :-) I just think about the amount of code being added just to add a few values that can be easily serialized/parsed from a standard JSON.

Reply:
Discussed offline. We leave i9t like that for now.